src/wallet/wallet.h

```diff
@@ -178,14 +178,20 @@ class ReserveDestination
 /** Address book data */
 class CAddressBookData
 {
+private:
+    std::string m_label;
 public:
-    std::string name;
+    const std::string& name;
```

Comment:
Why keep `name` around at all? I would prefer we get replace it with `m_label` and have a `Getlabel()` as well.

Reply:
Maybe as a follow-up refactor. Just trying to minimally fix the bug here. :)

Reply:
I agree that we should remove `name` in this PR. Leaving it here is confusing and makes the code less readable.

Reply:
It's also a pain to review because we can't just sed it...

Reply:
Ultimately, it's unrelated to this PR... All confusion/unreadability from it predates this.

Reply:
See here: https://github.com/bitcoin/bitcoin/commit/c16a57a2a9f694feaf0020e057fa9468e99045db

Reply:
Looks good. Please tag me when you open a PR for that branch.

Reply:
> Looks good. Please tag me when you open a PR for that branch.

I agree. Happy to review that change.